src/common/getConfigToUse.ts

```diff
@@ -1,28 +1,33 @@
 import fs from "fs";
 import path from "path";
 import { CONFIG_PATH } from "./paths";
-import type { RulesetResult } from "./types";
+import type { ExosConfig, RulesetResult, ExosScriptsNames } from "../common/types";
 
 /**
  * Retrieves the configuration to use.
- * @param configFileName The config file name
+ * @param scriptName The script name
  * @param defaultConfig The default config to use
  * @returns
  * If there is a custom configuration, we will call this file and send the default config.
  * Otherwise, we will simply return the default one.
  */
-function getConfigToUse<T>(configFileName: string, defaultConfig: T): RulesetResult<T> {
-  const customConfigPath: string = path.resolve(CONFIG_PATH, configFileName);
-  const customConfigExists: boolean = fs.existsSync(customConfigPath);
+function getConfigToUse<T>(scriptName: ExosScriptsNames, defaultConfig: T): RulesetResult<T> {
+  const exosConfigPath: string = path.resolve(CONFIG_PATH);
+  const exosConfigPathExists: boolean = fs.existsSync(exosConfigPath);
 
-  if (!customConfigExists) {
+  if (!exosConfigPathExists) {
     return { result: defaultConfig, isCustom: false };
   }
 
-  // We need to require this dynamically as the command is chosen by the user
-  // eslint-disable-next-line @typescript-eslint/no-var-requires
-  const customConfig = require(customConfigPath)(defaultConfig, { env: process.env.NODE_ENV });
-  return { isCustom: true, result: customConfig, customPath: customConfigPath };
+  const exosConfig = require(exosConfigPath) as ExosConfig;
+  const exosScriptCustomConfig = exosConfig.scripts && exosConfig.scripts[scriptName];
+
+  if (!exosScriptCustomConfig) {
+    return { result: defaultConfig, isCustom: false };
+  }
+
+  const customConfig = exosScriptCustomConfig(defaultConfig, { env: process.env.NODE_ENV });
```

Comment:
You are passing only NODE_ENV var and changing its name. Why is it needed? Maybe it will be better to pass `isProduction` boolean instead?

Reply:
I think we should pass both. The idea is to pass everything others need to know exactly what to do in the function.

Reply:
You can always access any env var through `process.env`. This additional arg looks redundant to me